src/lib/bcmp/bcmp_info.cpp

```diff
@@ -26,48 +25,34 @@ static BCMP_Linked_List_Generic INFO_REQUEST_LIST;
 static BmErr bcmp_send_info(void *dst) {
 
   BmErr err = BmENOMEM;
-  char *ver_str = (char *)bm_malloc(ver_str_max_len);
+  uint8_t version_string_len = strlen(version_string());
+  uint8_t device_name_len = strlen(device_name());
+  uint16_t info_len = sizeof(BcmpDeviceInfoReply) + version_string_len + device_name_len;
+  uint8_t *buf = (uint8_t *)bm_malloc(info_len);
 
-  if (ver_str) {
-    memset(ver_str, 0, ver_str_max_len);
+  if (buf) {
+    memset(buf, 0, info_len);
 
-    uint8_t ver_str_len =
-        snprintf(ver_str, ver_str_max_len, "%s@%s", APP_NAME, getFWVersionStr());
+    BcmpDeviceInfoReply *dev_info = (BcmpDeviceInfoReply *)buf;
```

Comment:
Rather than casting once to `uint8_t *` then casting again to `BcmpDeviceInfoReply *`, can we just cast the original alloc to `BcmpDeviceInfoReply *` from the beginning?